src/ProgressOnderwijsUtils/Data/SqlBatch.cs

```diff
@@ -44,14 +43,18 @@ public NonQuerySqlCommand WithTimeout(CommandTimeout timeout)
         public NonQuerySqlCommand(ParameterizedSql sql, CommandTimeout timeout)
             => (Sql, CommandTimeout) = (sql, timeout);
 
-        public int Execute(SqlConnection conn)
+        public void Execute(SqlConnection conn)
+            => Execute(conn, out _);
+
+        public void Execute(SqlConnection conn, out int nrOfRowsAffected)
         {
             using var cmd = this.ReusableCommand(conn);
             try {
                 if (string.IsNullOrWhiteSpace(cmd.Command.CommandText)) {
-                    return 0;
+                    nrOfRowsAffected = 0;
+                    return;
                 }
-                return cmd.Command.ExecuteNonQuery();
+                nrOfRowsAffected = cmd.Command.ExecuteNonQuery();

```

Comment:
Dit is nog beter dan `rowsAffected` in je andere  overigens; `rowsAffected` leest als een `boolean` ipv een `int`